### ISSUE TITLE
How to add support for other Viessmann devices (Viessmann Vitovalor PT2)?

### ISSUE THREAD

**Issue by vanThijs** [philipdaubmeier/GraphIoT#3]
Hi,

Firt of all thank you for your fantastic work!
I would like to add my Vitovalor PT2 (Fuelcell WKK) to my monitoring, but how can I find which parameters are available in the Viessmann API, so  I can add the fuelcell specific parameters?

With kind regards,
Thijs

**Comment by philipdaubmeier** [comment]
Thanks!
I do not have a Vitovalor, so I can neither see which features are available nor test it. Can you send me the (anonymized) API response for the device features? Then I can add it to the library and unit tests.

**Comment by vanThijs** [comment]
Can i use graphIOT to collect this information? I will try t this tonight.

**Comment by philipdaubmeier** [comment]
No, unfortunately you cant't find that out with GraphIoT, as it doesn't support the Vitovalor yet. But that is exactly what we can add with your help.

The easiest way would be that you go to https://vitoguide.viessmann.com/home and open the Developer Tools of your Browser. In Firefox for example you can open them with F12 and open the Network Tab. Then log into Vitoguide and click on your Vitovalor device. Check for a request to `/features` and look for the response payload. This should be a fairly long json string.

**Comment by philipdaubmeier** [comment]
Here is a screenshot how that should look like: `1` the Network Tab of the Developer Tools, `2` the request you have to look for, `3` the response tab where `4` the response payload can be found on the very bottom:

<img width="620" alt="screen" src="https://user-images.githubusercontent.com/1835730/78506897-2db9f380-777d-11ea-9321-52814a8e8d4f.png">

**Comment by vanThijs** [comment]
I have found it, i will send the json file now to the emailadres I found on your website.

**Comment by philipdaubmeier** [comment]
Added fuel cell related features in latest version.

**Comment by philipdaubmeier** [comment]
Now you can customize GraphIoT to make use of the new Vitovalor values - because I do not know which values exactly you want to store and look at. But this is quite easy for you to do - just fork the GraphIoT repo and make the following adjustments:

1. In [`ViessmannHeatingData.cs`](https://github.com/philipdaubmeier/GraphIoT/blob/master/src/GraphIoT.Viessmann/Database/ViessmannHeatingData.cs) add a database column for each value series you want to add and give it a name. Like:

```
[MaxLength(800)]
public string? MyFuelCellValueCurve { get; set; }
```

2. In [`ViessmannHeatingPollingService.cs`](https://github.com/philipdaubmeier/GraphIoT/blob/master/src/GraphIoT.Viessmann/Database/ViessmannHeatingPollingService.cs) add the new value to the `PollHeatingValue` method and to the `SaveHeatingValues` method

   You can find all new available methods here: [commit c4f88436781...](https://github.com/philipdaubmeier/GraphIoT/commit/c4f88436781cd688ef2867808fe6db4e30c5def9)

3. In [`ViessmannHeatingViewModel.cs`](https://github.com/philipdaubmeier/GraphIoT/blob/master/src/GraphIoT.Viessmann/Database/ViessmannHeatingViewModel.cs) add the new value series to the switch statement of the `Graph` method, give them a display name and unit.
4. Create a new entity framework code-first migration by running `Add-Migration "vitovalor_v1"` in the nuget package manager console of Visual Studio.
5. If you compile and run (or deploy) everything, the new values should pop up automatically in Grafana when creating/editing a dashboard
